### ISSUE TITLE
Multiple types per source

### ISSUE THREAD

**Issue by bhousel** [osmlab/editor-layer-index#179]
Right now we have the [type](https://github.com/osmlab/editor-layer-index/blob/b5cce1d9a9ca92af5701553286dee86268de63c1/schema.json#L11-L14) property, which is one of `"tms", "wms", "bing", "scanex"`.

It would be nice to have each source file include all of the supported types for that source, rather than duplicating sources.

e.g.
```js
source_types: {
  tms: { /*stuff*/ },
  wms: { /*other stuff*/ }
}
```

**Comment by iandees** [comment]
That makes sense to me.

**Comment by simonpoole** [comment]
IMHO a good idea, but not backwards compatible and given that the format isn't versioned simply changing it would not be nice (as with OSM I assume that we don't really know who is consuming the data, if anybody).

**Comment by tyrasd** [comment]
related discussion: #24

**Comment by bhousel** [comment]
> IMHO a good idea, but not backwards compatible and given that the format isn't versioned simply changing it would not be nice (as with OSM I assume that we don't really know who is consuming the data, if anybody).

after #168 versioning the project becomes automatic

**Comment by simonpoole** [comment]
Well except that my fork will continue to use the old scripts etc.

**Comment by iandees** [comment]
How does switching to Node make versioning automatic?

**Comment by bhousel** [comment]
> How does switching to Node make versioning automatic?

We publish regular releases to npm and follow semver..

**Comment by iandees** [comment]
Gotcha. We can do that with Python, too, so I was checking if there was something else.

**Comment by simon04** [comment]
JOSM uses the notion of `mirror`s, see https://josm.openstreetmap.de/changeset/7990/josm for the change in the XSD file, and https://josm.openstreetmap.de/wiki/Maps/Italy#SouthTyrolTopomap for a layer specified as `wms`+`tms`.

**Comment by grischard** [comment]
We also duplicate a lot of polygons - it would be good if we could define one geojson feature, then multiple sources per feature, then multiple types per source.

**Comment by tyrasd** [comment]
> We also duplicate a lot of polygons

An alternative solution to that sub-problem would be to distribute the data (also) as [topojson](https://github.com/topojson/topojson) format, which implicitly gets rid of duplicate outline segments.

**Comment by Marc-marc-marc** [comment]
some layers also have mirror url for the same type (for ex a mirror done by a local chapter to use when upstream is down). it would also be usefull to have them in the same layer
